extension/observer/k8sobserver/ingress_endpoint.go

```diff
@@ -0,0 +1,103 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package k8sobserver // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer/k8sobserver"
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	v1 "k8s.io/api/networking/v1"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer"
+)
+
+// convertIngressToEndpoints converts a ingress instance into a slice of endpoints. The endpoints
+// include an endpoint for each path that is mapped to an ingress.
+func convertIngressToEndpoints(idNamespace string, ingress *v1.Ingress) []observer.Endpoint {
+	endpoints := []observer.Endpoint{}
+
+	// Loop through every ingress rule to get every defined path.
+	for _, rule := range ingress.Spec.Rules {
+		scheme := getScheme(rule.Host, getTLSHosts(ingress))
+
+		if rule.HTTP != nil {
+			// Create endpoint for each ingress rule.
+			for _, path := range rule.HTTP.Paths {
+				endpointID := observer.EndpointID(fmt.Sprintf("%s/%s/%s%s", idNamespace, ingress.UID, rule.Host, path.Path))
```

Comment:
I think it makes sense to escape the backslashes here and be explicit with the ID format to avoid any kind of unlikely ID collision? 
```suggestion
				endpointID := observer.EndpointID(fmt.Sprintf("%s/%s/%s/%s", idNamespace, ingress.UID, rule.Host, escBackslashes(path.Path)))
```

Reply:
I don't know; that was my initial question. How do other components deal with that ?